pkg/apis/core/validation/shoot.go

```diff
@@ -1083,7 +1083,8 @@ func validateEncryptionConfig(encryptionConfig *core.EncryptionConfig, version s
 	seenResources := sets.New[string]()
 	for i, resource := range encryptionConfig.Resources {
 		idxPath := fldPath.Child("encryptionConfig", "resources").Index(i)
-		if seenResources.Has(resource) {
+		// core resources can be mentioned with empty group (eg: secrets.)
+		if seenResources.Has(resource) || seenResources.Has(strings.TrimSuffix(resource, ".")) {
```

Comment:
```suggestion
		if seenResources.Has(strings.TrimSuffix(resource, ".")) {
```

Reply:
This won't work if the list has `<resource>.` twice. However it should work for the check below. Adapted it in https://github.com/gardener/gardener/pull/8964/commits/7eb28107806c251c1b980bf582c7e084d1e032c3